mozilla/fxa
# Integrate 2FA setup option into login flow

If the RP requests 2FA, but the user does not have 2FA enabled, this change will send the user into a 2FA setup flow after signing in.

How to test out the flow locally:
* set up a test account
* go to 123done and click the "sign in 2FA required" button in the header
* you'll sign in, then be taken into a 2FA setup flow
* after successful 2FA setup, you'll be redirected to 123done

The code has a few open TODOs and I still need to wrap up copy changes and tests, but it's far enough along to ask for review on the bulk of the work. Probably the biggest open question is how to handle the user canceling--probably just redirect back to the RP with 2FA not enabled, I guess?

For lots more context, including screenshots of the settings and login flow versions of the 2FA setup flow, see this github comment: https://github.com/mozilla/fxa/issues/5089#issuecomment-619297409
